packages/react-text/Spec.md

```diff
@@ -0,0 +1,152 @@
+| Set as disabled Text component                                                                                   |         | disabled      | Missing in Fabric    |
+| Set as error Text component                                                                                      |         | error         | Missing in Fabric    |
+| The text can appear more important and draw user's attention                                                     |         | important     | Missing in Fabric    |
+| Set as success Text component                                                                                    |         | success       | Missing in Fabric    |
+| The text can signify a temporary state                                                                           |         | temporary     | Missing in Fabric    |
+| Set as timestamp Text component                                                                                  |         | timestamp     | Missing in Fabric    |
+| Override for theme site variables to allow modifications of component styling via themes.                        |         | variables     | Missing in Fabric    |
+| The weight for the Text component                                                                                |         | weight        | Missing in Fabric    |
+| Additional CSS class name(s) to apply.                                                                           |         | className     | Missing in Fabric    |
+| Additional CSS styles to apply to the component instance.                                                        |         | styles        | Missing in Fabric    |
+| -                                                                                                                |         | design        | Missing in Fabric    |
+
+## API Proposal
+
+The new Text will provide props to customize text according to the standards defined by Fluent design. The component will not be tied down to specific application usage, like Stardust did, for example, with `mention` and `timestamp` props, and will allow a bigger degree of freedom when customizing, unlike we did in Fabric.
+We're also introducing a new concept of wrappers for the main design variants (i.e. 'Title', 'Subtitle', 'Caption') to improve readability and semantics of the code. The wrappers are expected to have fixed styles in terms of size and font family, but flexible for the other supported props in Text.
+These wrappers follow the Fluent UI language so for any deviation, regarding the fixed styles mentioned above, should use the Text component instead, given that this is fully customizable.
+
+| Property      | Type                                                       | Default value | Comments |
+| ------------- | ---------------------------------------------------------- | ------------- | -------- |
+| as            | keyof JSX.IntrinsicElements                                | "span"        |
```

Comment:
I am not sure that this was discussed, but I suggest to restrict this set to `p`, `span`, etc. It's aligned with a strategy that we agreed in #17434. For example, I don't consider usages below as valid:

```tsx
<>
  <Text as="table" />
  <Text as="ul" />
</>
```